### ISSUE TITLE
errors and warnings in /dev/

### ISSUE THREAD

**Issue by torleif** [silverstripe/silverstripe-graphql-devtools#5]
navigating to http://localhost:8888/dev/:

cms/ framework 4.0-alpha-7.0

```[Notice] Undefined index: links
GET /dev/?flushtoken=2096d9f58a0906c73223e954a9198eb9&flush=1
Line 144 in /Applications/MAMP/htdocs/framework/src/Dev/DevelopmentAdmin.php
```
and 

```
[Warning] Invalid argument supplied for foreach()
GET /dev/?flushtoken=2096d9f58a0906c73223e954a9198eb9&flush=1
Line 144 in /Applications/MAMP/htdocs/framework/src/Dev/DevelopmentAdmin.php
```

**Comment by dhensby** [comment]
I can replicate this and it also happens on latest 4.0.x-dev.

Replicated by doing the following:

```sh
$ composer create-project silverstripe/installer:4.0.0-alpha7 testsite
$ cd testsite
$ composer require silverstripe/graphql-devtools
```

full backtrace:

```
ERROR [Notice]: Undefined index: links
IN GET /dev/?flushtoken=f93e025cee7392f58579a48903ec2bdb&ajax=1&flush=
Line 144 in /vagrant/www/framework/src/Dev/DevelopmentAdmin.php

Source
======
  135:     /**
  136:      * @return array of url => description
  137:      */
  138:     protected static function get_links()
  139:     {
  140:         $links = array();
  141: 
  142:         $reg = Config::inst()->get(__CLASS__, 'registered_controllers');
  143:         foreach ($reg as $registeredController) {
* 144:             foreach ($registeredController['links'] as $url => $desc) {
  145:                 $links[$url] = $desc;
  146:             }
  147:         }
  148:         return $links;
  149:     }
  150: 

Trace
=====
Monolog\ErrorHandler->handleError(8,Undefined index: links,/vagrant/www/framework/src/Dev/DevelopmentAdmin.php,144,Array)
DevelopmentAdmin.php:144

SilverStripe\Dev\DevelopmentAdmin::get_links()
DevelopmentAdmin.php:86

SilverStripe\Dev\DevelopmentAdmin->index(SilverStripe\Control\HTTPRequest)
RequestHandler.php:318

SilverStripe\Control\RequestHandler->handleAction(SilverStripe\Control\HTTPRequest,index)
Controller.php:282

SilverStripe\Control\Controller->handleAction(SilverStripe\Control\HTTPRequest,index)
RequestHandler.php:217

SilverStripe\Control\RequestHandler->handleRequest(SilverStripe\Control\HTTPRequest,SilverStripe\ORM\DataModel)
Controller.php:211

SilverStripe\Control\Controller->handleRequest(SilverStripe\Control\HTTPRequest,SilverStripe\ORM\DataModel)
Director.php:435

SilverStripe\Control\Director::handleRequest(SilverStripe\Control\HTTPRequest,SilverStripe\Control\Session,SilverStripe\ORM\DataModel)
Director.php:184

SilverStripe\Control\Director::direct(/dev/,SilverStripe\ORM\DataModel)
main.php:226

ERROR [Warning]: Invalid argument supplied for foreach()
IN GET /dev/?flushtoken=f93e025cee7392f58579a48903ec2bdb&ajax=1&flush=
Line 144 in /vagrant/www/framework/src/Dev/DevelopmentAdmin.php

Source
======
  135:     /**
  136:      * @return array of url => description
  137:      */
  138:     protected static function get_links()
  139:     {
  140:         $links = array();
  141: 
  142:         $reg = Config::inst()->get(__CLASS__, 'registered_controllers');
  143:         foreach ($reg as $registeredController) {
* 144:             foreach ($registeredController['links'] as $url => $desc) {
  145:                 $links[$url] = $desc;
  146:             }
  147:         }
  148:         return $links;
  149:     }
  150: 

Trace
=====
Monolog\ErrorHandler->handleError(2,Invalid argument supplied for foreach(),/vagrant/www/framework/src/Dev/DevelopmentAdmin.php,144,Array)
DevelopmentAdmin.php:144

SilverStripe\Dev\DevelopmentAdmin::get_links()
DevelopmentAdmin.php:86

SilverStripe\Dev\DevelopmentAdmin->index(SilverStripe\Control\HTTPRequest)
RequestHandler.php:318

SilverStripe\Control\RequestHandler->handleAction(SilverStripe\Control\HTTPRequest,index)
Controller.php:282

SilverStripe\Control\Controller->handleAction(SilverStripe\Control\HTTPRequest,index)
RequestHandler.php:217

SilverStripe\Control\RequestHandler->handleRequest(SilverStripe\Control\HTTPRequest,SilverStripe\ORM\DataModel)
Controller.php:211

SilverStripe\Control\Controller->handleRequest(SilverStripe\Control\HTTPRequest,SilverStripe\ORM\DataModel)
Director.php:435

SilverStripe\Control\Director::handleRequest(SilverStripe\Control\HTTPRequest,SilverStripe\Control\Session,SilverStripe\ORM\DataModel)
Director.php:184

SilverStripe\Control\Director::direct(/dev/,SilverStripe\ORM\DataModel)
main.php:226
```

**Comment by gorriecoe** [comment]
PR https://github.com/silverstripe/silverstripe-framework/pull/8033#issuecomment-384459987 should fix this but could someone merge #7 or #8.  Sorry I didn't notice #7 but I'll keep #8 to keep attention on this issue.

**Comment by senorgeno** [comment]
Hey can we get this merged in?

**Comment by gorriecoe** [comment]
This has been fixed from the frameworks end from 4.2 but I would still like this merged here aswell.

**Comment by chillu** [comment]
Sorry for the delay, but please fix this by upgrading to 4.2+

**Comment by gorriecoe** [comment]
Can we still get either #7 or #8 merged in?  It will be handy to have the link available via dev.
